engine/src/main/java/io/camunda/zeebe/engine/processing/timer/TriggerTimerProcessor.java

```diff
@@ -160,8 +164,12 @@ private void rescheduleTimer(
       repetitions--;
     }
 
-    final Interval interval = timer.map(Timer::getInterval).get();
-    final Timer repeatingInterval = new RepeatingInterval(repetitions, interval);
+    final Interval refreshedInterval =
+        timer
+            .map(Timer::getInterval)
+            .map(interval -> refreshInterval(record.getDueDate(), interval))
```

Comment:
Excellent idea :+1: 

Since you use the last due date of the timer instead of the current time, we can avoid a time shift. So, the timer is triggered at the same time (and not always a bit later).

:wrench: Please add a comment to make clear why we use the due date.